docs/latest/custom-tools/creating-your-tool.md

```diff
@@ -0,0 +1,39 @@
+## Creating Your Tool {#creating-your-tool}
+
+Once you have an appropriate base class chosen (we will use the `BaseTool` in this example), you can extend it to start building your tool. In this example we shall make a tool that logs 'Hello cornerstoneTools!' to the console on every mouse click.
+
+### Class Definition
+
+By convention the class name should be in PascalCase, and suffixed with:
+- `Tool` - If it supports both `mouse` and `touch` input.
+- `MouseTool` - If it only supports `mouse` input.
+- `TouchTool` - If it only supports `touch` input.
+
+For example, our example is going to support mouse, so we shall call it the `HelloWorldMouseTool`:
+
+```js
+import external from './../externalModules.js';
+import BaseTool from './../base/BaseTool.js';
```

Comment:
The imports only work internally. We need to identify the best way to access, extend/implement these interfaces externally.

Reply:
Quilljs has an interesting approach:

_Example of extending a "base class"_
```js
// https://github.com/quilljs/parchment/#blots
import Quill from 'quill'
let Inline = Quill.import('blots/inline')

class SpellingBlot extends Inline {
  static create(value) {
    let node = super.create(value)
    node.setAttribute('data-invalid-spelling', '')
    node.setAttribute('data-alternative-spellings', [])
    return node
  }

  static formats(domNode) {
    return {
      isInvalidSpelling: domNode.hasAttribute('data-invalid-spelling'),
      alternativeSpellings: domNode.getAttribute('data-alternative-spellings'),
    }
  }
}

SpellingBlot.blotName = 'spelling'
SpellingBlot.className = 'spelling'
SpellingBlot.tagName = 'SPAN'

export default SpellingBlot
```

And then to add it... [register](https://quilljs.com/docs/api/#register)
```js
Quill.register(SpellingBlot, true);
```

Reply:
Yeah, we need to expose a load of this stuff. So far the `BaseTool`s aren't even exposed at the surface level. I'm using the doc writing as a way of identifying what'd be needed.